components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -0,0 +1,385 @@
+#ifndef SCREAM_SCORPIO_INPUT_HPP
+#define SCREAM_SCORPIO_INPUT_HPP
+
+#include "scream_config.h"
+#include "ekat/ekat_parameter_list.hpp"
+#include "ekat/std_meta/ekat_std_enable_shared_from_this.hpp"
+#include "ekat/mpi/ekat_comm.hpp"
+#include "ekat/std_meta/ekat_std_utils.hpp"
+#include "ekat/ekat_parse_yaml_file.hpp"
+
+#include "share/io/scream_scorpio_interface.hpp"
+
+#include "share/field/field_repository.hpp"
+#include "share/field/field_header.hpp"
+#include "share/field/field.hpp"
+#include "share/field/field_identifier.hpp"
+
+#include "share/grid/grids_manager.hpp"
+/*  The AtmosphereInput class handles all input streams to SCREAM.
+ *  It is important to note that there does not exist an InputManager,
+ *  like in the case of output.  So all input streams have to be managed
+ *  directly by the process that requires it.
+ *
+ *  The typical input call will be to the outward facing routine 'pull_input'.
+ *
+ *  pullinput calls the three typical init, run and finalize routines in order
```

Comment:
Consider moving this comment to the method declaration in the class, and possibly describing the other methods there.